specification-discussions/20200309-dx-mosaic-testnet-gen1.md

```diff
@@ -0,0 +1,27 @@
+---
+title: "DX for Mosaic testnet gen-1"
+disqus: https://hackmd.io/gTarrQN9QfusSvZeZp2W_w
+---
+
+# DX for Mosaic testnet gen-1
+
+| Version | Last Updated | Component                   |
+| ------- | ------------ | --------------------------- |
+| 0.14    | 09/03/2020   | DX for Mosaic testnet gen-1 |
+
+## User stories
+
+1. As a developer, I should be able to see which metachains exist for goerli (as origin chain)
+2. As a developer, I should be able to see what the base token is for each metachain (Ost or GoEth)
+3. As a developer, I should be able to connect to a public web3 RPC for the origin chain
+4. As a developer, I should be able to connect to a public web3 RPC for a metachain
+5. As a developer, I should be able to get the base token on the origin (GoEth; from faucet)
+6. As a developer, I should be able to get the base token on a metachain (Ost or GoEth; from faucet)
+7. As a developer, I should be able to get Ost ERC20 on the origin chain (from faucet)
+8. As a developer, I should be able to view total deposited ERC20 balances (for all ERC20's for all metachains)
+9. As a user, I should be able to deposit wGoEth, Ost, or any ERC20 into an ERC20gateway on, and receive it as an ERC20 on the metachain (handled by any facilitator)
+10. As a user, I should be able to withdraw any ERC20(not base token) from the ERC20Cogateway and receive the corresponding wGoEth, Ost, or ERC20 token on the origin
+11. As a user, I should be able to deposit the ERC20(base token) into the ConsensusGateway and receive them as base token on metachain
+12. As a user, I should be able to withdraw the ERC20(base token) from the ConsensusCogateway and receive them as base token on the origin chain
+13. As a developer, I should be able to deploy any contracts to the metachain
```

Comment:
add user story

```
As a user, I should be able to send any valid transactions to any metachain.
```

this is indirectly satisfied from 11. (assuming default Ethereum behaviour is not broken), but worthwhile explicitly mentioning i think

Reply:
done

Reply:
why narrow it to `as a developer`?

Reply:
Updated 13 now, as it can be included with `As a user, I should be able to send any valid transactions to any metachain.` and be more generalised